crates/biome_service/src/file_handlers/graphql.rs

```diff
@@ -147,6 +166,117 @@ impl ServiceLanguage for GraphqlLanguage {
             .with_file_path(path.as_path())
             .with_suppression_reason(suppression_reason)
     }
+
+    fn formatter_enabled_for_this_file_path(settings: Option<&Settings>, path: &Utf8Path) -> bool {
+        settings
+            .and_then(|settings| {
+                let overrides_activity =
+                    settings
+                        .override_settings
+                        .patterns
+                        .iter()
+                        .rev()
+                        .find_map(|pattern| {
+                            check_feature_activity(
+                                pattern.languages.graphql.formatter.enabled,
+                                pattern.formatter.enabled,
+                                true,
+                            )
+                            .and_then(|enabled| {
+                                // Then check whether the path satisfies
+                                if pattern.include.matches_path(path)
+                                    && !pattern.exclude.matches_path(path)
+                                {
+                                    Some(enabled)
+                                } else {
+                                    None
+                                }
+                            })
+                        });
+
+                overrides_activity.or(check_feature_activity(
+                    settings.languages.graphql.formatter.enabled,
+                    settings.formatter.enabled,
+                    false,
+                ))
+            })
+            .unwrap_or_default()
+            .into()
+    }
```

Comment:
These functions are quite a bit of a repeated pattern. Every language has three of these and it seems the only things that really changes are the `pattern.languages.graphql.formatter.enabled` and `pattern.formatter.enabled` fields. I think it would be good to create a helper for this, so they don't need to be copy-pasted as much.

I suspect we can even achieve this by moving the bodies of these functions fully onto the `ServiceLanguage` trait, and instead let the language-specific implementations implement getters that return fields such as `settings.languages.graphql.formatter.enabled`.

Reply:
I already explored this solution, but unfortunately, the compiler gets in the way because we would need to define tree const generics bool, and eventually, the last `check_feature_activity` doesn't compile because the compiler expects a different type.

I don't have enough knowledge and energies to go back to the previous solution and fix the compiler errors.

I'd prefer to keep this here now, and maybe explore a solution later.